pkg/dab/module_test.go

```diff
@@ -0,0 +1,175 @@
+package dab_test
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	qt "github.com/frankban/quicktest"
+
+	"github.com/warptools/warpforge/pkg/dab"
+	"github.com/warptools/warpforge/wfapi"
+)
+
+var veryLongDomain = strings.Repeat("a", 63) + "." + strings.Repeat("b", 63) + "." + strings.Repeat("c", 63) + "." + strings.Repeat("d", 61)
+var manySubdomains = strings.Repeat("a.", 126) + "b"
+
+func TestValidateModuleName(t *testing.T) {
+	qt.Assert(t, veryLongDomain, qt.HasLen, 253)
+	qt.Assert(t, manySubdomains, qt.HasLen, 253)
+	for _, testcase := range []struct {
+		name    string // if left empty will use the value name
+		value   string
+		checker qt.Checker
+	}{
+		// good names | happy path
+		{"", "example", qt.IsNil},
+		{"", "example.com", qt.IsNil},
+		{"", "example.com/b", qt.IsNil},
+		{"", "a.b/c", qt.IsNil},
+		{"", "a.b.c", qt.IsNil},
+		{"", "abc.def.ghi/jkl", qt.IsNil},
+		{"", "abc-def.ghi/jkl", qt.IsNil},
+		{"", "example.com/b/c", qt.IsNil},
+		{"", "1.2", qt.IsNil},
+		{"", "1.2/3", qt.IsNil},
+		{"", "1.2/3/4", qt.IsNil},
+		{"", "a.b/foo", qt.IsNil},
+		{"", "a.b/foobar", qt.IsNil},
+		{"", "a.b/foo/bar", qt.IsNil},
+		// underscore considered for valid chars but removed
+		{"", "abc_def.ghi/jkl", qt.IsNotNil},
+		{"", "a.b/foo_bar", qt.IsNotNil},
```

Comment:
I wouldn't mind putting underscore into the middle grouping of `validation_moduleNamePathHunk_regexpStr`, which would make `"a.b/foo_bar"` valid.  I don't feel strongly about it, but I feel like restricting that char might produce more headdesk than not.

(I think we already have plenty of space reserved for all other purposes, e.g. the filesystem projection of catalogs, already from reserving leading underscore alone.)